### PR TITLE
Adds Google service account "noop"s for "mobile-dep"

### PR DESCRIPTION
**Pull request by mitchhentges**
As discussed in #mozbuild IRC channel a couple minutes ago.
Alternatively, I can:
* Leave the strings blank rather than having "noop"
* Reuse the same structure as "dep", including reusing `$google_play_config['dep']`
* Reuse the same structure as "dep" but fill out the contents of `$google_play_config['mobile-dep']`

:thinking: up to you my guy

### CODE REVIEW

**Comment by mitchhentges** [comment]
Note: I also added `do_not_contact_google_play` option (thought this was a `task` payload option, but it's actually in the instance config!)

File: modules/pushapk_scriptworker/manifests/settings.pp [hunk 1]
```diff
@@ -12,58 +12,63 @@
...
-        verbose_logging          => true,
-        taskcluster_client_id    => secret('pushapk_scriptworker_taskcluster_client_id_dep'),
-        taskcluster_access_token => secret('pushapk_scriptworker_taskcluster_access_token_dep'),
-        scope_prefix             => 'project:releng:googleplay:',
-        cot_product              => 'firefox',
-
-        sign_chain_of_trust      => false,
-        verify_chain_of_trust    => true,
-        verify_cot_signature     => false,
+        worker_group               => 'dep-pushapk',
```
Comment (mitchhentges) [inline modules/pushapk_scriptworker/manifests/settings.pp]: Sorry for the indentation change, `do_not_contact_google_play` is just a super fat string :annoyed:

File: modules/pushapk_scriptworker/manifests/settings.pp [hunk 2]
```diff
@@ -12,58 +12,63 @@
...
+        worker_type                => 'dep-pushapk',
+        verbose_logging            => true,
+        taskcluster_client_id      => secret('pushapk_scriptworker_taskcluster_client_id_dep'),
+        taskcluster_access_token   => secret('pushapk_scriptworker_taskcluster_access_token_dep'),
+        scope_prefix               => 'project:releng:googleplay:',
```
[the remark this answers is not in the document]
Comment (mitchhentges) [reply modules/pushapk_scriptworker/manifests/settings.pp]: I don't quite understand the incorrect indentation here: I've followed the existing pattern of indentation that the file had before I edited it (`$_env_configs` is indented with two spaces, everything is indented by 4). All the `=>` indents are aligned - can you elaborate?

File: modules/pushapk_scriptworker/manifests/settings.pp [hunk 3]
```diff
@@ -146,6 +151,24 @@
               'release' => 'release',
             }
         }
+        'mobile-dep': {
+            $google_play_config = {
+                'dep'  => {
+                    service_account             => 'noop',
```
[the remark this answers is not in the document]
Comment (mitchhentges) [reply modules/pushapk_scriptworker/manifests/settings.pp]: > Good idea on putting "noop"! Could you do the same at https://github.com/mozilla-releng/build-puppet/pull/317/files#diff-aa7dacd4373035e75425f8728aca336dR101 ?

Great call here, I just found:
```
    dep-pushapkworker-1.srv.releng.use1.mozilla.com:
        dep:
            service_account: 'dummy'
            certificate: 'dummy'
```

in `secrets.eyaml` on the puppet master. I'll hard-code that, so it's more obvious, and re-use it on `mobile-dep`

File: modules/pushapk_scriptworker/manifests/settings.pp [hunk 4]
```diff
@@ -146,6 +151,24 @@
...
+                    certificate                 => 'noop',
+                    certificate_target_location => 'noop',
```
[the remark this answers is not in the document]
Comment (mitchhentges) [reply modules/pushapk_scriptworker/manifests/settings.pp]: Looks like `dep` uses `"${root}/dep.p12"`, so we have a winner :+1:

**Comment by mitchhentges** [comment]
Thanks for the notes, I've implemented them with the help of `g-k` and Aki :)
Note: once this is merged, the `dep` part in `secrets.eyaml` can be removed, since all that it was was dummy values (which have now been pulled into this config and inlined to make their purpose more clear)

File: modules/pushapk_scriptworker/manifests/settings.pp [hunk 5]
```diff
@@ -191,6 +212,6 @@
         'jarsigner_key_store' => $jarsigner_keystore,
         'jarsigner_certificate_aliases' => $jarsigner_certificate_aliases_content,
 
-        'taskcluster_scope_prefix' => $_env_config['scope_prefix'],
+        'taskcluster_scope_prefix' => $_env_config['scope_prefix']
```
Comment (JohanLorenzo) [inline modules/pushapk_scriptworker/manifests/settings.pp]: Nit: No need to strip the comma at the end. Hiera hashes are smarter than JSON objects.

File: modules/pushapk_scriptworker/manifests/settings.pp (hunk 2 above)
[the remark this answers is not in the document]
Comment (JohanLorenzo) [reply modules/pushapk_scriptworker/manifests/settings.pp]: The indent was bad before your patch. In puppet, arrows (of the same hash) are usually aligned one with another - like line 17 and 18. Anyway, that's not a big deal as the issue was already here in the first place.

File: modules/pushapk_scriptworker/manifests/settings.pp [hunk 6]
```diff
@@ -93,15 +93,15 @@
         'dep': {
             $google_play_config = {
                 'dep'  => {
-                    service_account             => $_google_play_accounts['dep']['service_account'],
-                    certificate                 => $_google_play_accounts['dep']['certificate'],
+                    service_account             => 'dummy',
```
Comment (JohanLorenzo) [inline modules/pushapk_scriptworker/manifests/settings.pp]: Thanks for the clean up! :+1:

**Comment by MihaiTabara** [comment]
Ever since we merged this, we get alerts like: 

Wed 10:51:25 UTC [8003] [Unknown] mobil-pushapkworker-1.srv.releng.use1.mozilla.com:Scriptworker gpg rebuild log age is CRITICAL: FILE_AGE CRITICAL: /builds/scriptworker/logs/rebuild_gpg_homedirs.log is 7h 18m 23s seconds old and 0 bytes (http://m.mozilla.org/Scriptworker+gpg+rebuild+log+age)

We might need to nuke the gpg home dirs or so.
@JohanLorenzo @mitchhentges are you okay with me doing so? Seems like this machine is used by focus and reference-browser so I guess it's still minimal impact.

**Comment by escapewindow** [comment]
Did that fix it? I was able to ssh in and run `rebuild_gpg_homedirs` manually, but the warning came back (I pinged you at this point).

**Comment by MihaiTabara** [comment]
Eventually I ended up nuking the entire `/builds/scriptworker` folder which silenced all the warnings errors 👍

**Pull request merged**